hbase-examples/src/main/cpp/DemoClient.cpp

```diff
@@ -79,15 +78,15 @@ main(int argc, char** argv)
     return -1;
   }
   bool isFramed = false;
-  boost::shared_ptr<TTransport> socket(new TSocket(argv[1], boost::lexical_cast<int>(argv[2])));
-  boost::shared_ptr<TTransport> transport;
+  std::shared_ptr<TTransport> socket(std::make_shared<TSocket>(argv[1], std::stoi(argv[2])));
```

Comment:
I think you mean socket = make_shared<TSocket>.....

With your code, I think it calls into the move constructor..